src/js/common/schemaform/Address.jsx

```diff
@@ -0,0 +1,275 @@
+import _ from 'lodash';
+import { set } from 'lodash/fp';
+
+import ErrorableSelect from '../components/form-elements/ErrorableSelect';
+import ErrorableTextInput from '../components/form-elements/ErrorableTextInput';
+// import { isNotBlank, isBlankAddress, validateIfDirty, isValidUSZipCode, isValidCanPostalCode } from '../utils/validations';
+import { countries, states } from '../utils/options-for-select';
+
+/**
+ * Input component for an address.
+ *
+ * No validation is provided through a currently stubbed isAddressValid function.
+ */
+class Address extends React.Component {
+  constructor(props) {
+    super();
+    this.handleChange = this.handleChange.bind(this);
+    this.handleBlur = this.handleBlur.bind(this);
+    // this.validateAddressField = this.validateAddressField.bind(this);
+    // this.validatePostalCode = this.validatePostalCode.bind(this);
+    this.state = { value: props.formData, touched: { street: false, city: false, state: false, country: false, postalCode: false } };
+  }
+
+  componentWillMount() {
+    this.id = _.uniqueId('address-input-');
+  }
+
+  handleBlur(field) {
+    const newState = _.set(['touched', field], true, this.state);
+    this.setState(newState, () => {
+      if (newState.touched.street && newState.touched.city && newState.touched.state && newState.touched.country && newState.touched.postalCode) {
+        this.props.onBlur(this.props.id, newState.value);
+      }
+    });
+  }
+
+  // TODO: Look into if this is the best way to update address,
+  // it is incredibly slow right now
+  handleChange(path, update) {
+    let newState = _.set(this.state, ['value', path], update);
+    newState = _.set(newState, ['touched', path], true);
+    
+    // if country is changing we should clear the state
+    if (path === 'country') {
+      newState = _.set(newState, ['value', 'state'], '');
+    }
+
+    this.props.onChange(newState.value);
+  }
+
+  // validateAddressField(field) {
+  //   if (this.props.required || !isBlankAddress(this.props.value)) {
+  //     return validateIfDirty(field, isNotBlank);
+  //   }
+
+  //   return true;
+  // }
+
+  // validatePostalCode(postalCodeField) {
+  //   let isValid = true;
+
+  //   if (this.props.required || !isBlankAddress(this.props.formData)) {
+  //     isValid = isValid && validateIfDirty(postalCodeField, isNotBlank);
+  //   }
+
+  //   if (this.props.formData.country === 'USA' && isNotBlank(postalCodeField.value)) {
+  //     isValid = isValid && validateIfDirty(postalCodeField, isValidUSZipCode);
+  //   }
+
+  //   if (this.props.formData.country === 'CAN' && isNotBlank(postalCodeField.value)) {
+  //     isValid = isValid && validateIfDirty(postalCodeField, isValidCanPostalCode);
+  //   }
+
+  //   return isValid;
+  // }
+
+  isMilitaryCity(city) {
+    const lowerCity = city.toLowerCase().trim();
+
+    return lowerCity === 'apo' || lowerCity === 'fpo' || lowerCity === 'dpo';
+  }
+
+  render() {
+    let stateList = [];
+
+    const states = {
```

Comment:
For whatever reason, even though I'm importing `states` from `utils/options-for-select`, it wasn't registered as actually being imported. For a shortcut to get this working for now, just copied and pasted the states list here, but will address a more permanent fix later.